src/utils/StorageService.ts

```diff
@@ -0,0 +1,143 @@
+import * as path from "path";
+import * as fs from "fs";
+import { MarketsEmission } from "../ages/distributions/MarketsEmission";
+import { Proof, Proofs } from "../ages/distributions/Proofs";
+import { numberOfEpochs } from "../ages/ages";
+import { UsersDistribution } from "../ages/distributions/UsersDistribution";
+
+export interface StorageService {
+  readMarketDistribution: (age: string, epoch: string) => Promise<MarketsEmission | undefined>;
+  writeMarketEmission: (age: string, epoch: string, emission: MarketsEmission) => Promise<void>;
+  readUsersDistribution: (age: string, epoch: string) => Promise<UsersDistribution | undefined>;
+  writeUsersDistribution: (age: string, epoch: string, distribution: UsersDistribution) => Promise<void>;
+  readProofs: (epoch: number) => Promise<Proofs | undefined>;
+  readAllProofs: () => Promise<Proofs[]>;
+  readUserProof: (epoch: number, address: string) => Promise<Proof | undefined>;
+  writeProofs: (epoch: number, proofs: Proofs) => Promise<void>;
+}
+
+export type ProofsCache = { [epoch: number]: Proofs | undefined };
+export type MarketsEmissionCache = {
+  [age: string]: {
+    [epoch: string]: MarketsEmission | undefined;
+  };
+};
+export type UsersDistributionCache = {
+  [age: string]: {
+    [epoch: string]: UsersDistribution | undefined;
+  };
+};
+
+export class FileSystemStorageService implements StorageService {
+  #emissionsCache: MarketsEmissionCache = {};
+  #distributionsCache: UsersDistributionCache = {};
+  #proofsCache: ProofsCache = {};
+  #distributionRoot = "../../distribution";
```

Comment:
This is not always the case and it can depend on the depth of the file where storage is used. I suggesst to use `fs` instead of `require` for this reason, assuming that `FileSystemStorageService` is always used in a Node environment. 
What do you think?

Reply:
I don't get why it's not always the case, everything becomes relative to `__dirname` of `StorageService`, so everything becomes `../../distribution`.
We can switch to `fs` if you want, I just don't really see the differences between the two.

Reply:
oh ok got it, was expecting the path to start where the object was instanced.